APPS 4242 Add glue test and update docs (#37)
spryker-sdk/spryk-src: docs/spryk_configuration_reference.md

### mode
The mode of the Spryk. It used for running the specific Spryks by passing the `--mode` option into the command. The reserved `both` value allows to run Spryk in any case.

`mode: project` - Spryk will be run only with `--mode=project` option in CLI command.

`mode: core` - Spryk will be run only with `--mode=core` option in CLI command.

`mode: both` - Spryk will be run regardless the `--mode` option value or without this option at all in CLI command.

### level
Is used only for Spryk dumper to dump the specific level of Spryks
`vendor/bin/spryk-dump --level=1` or `vendor/bin/spryk-dump --level=all`.

### condition
Defines the condition of the Spryk execution. If condition evaluates to false the Spryk execution will be skipped with it's (pre)postSpryks. The arguments that are used in condition must be defined in Spryk arguments list.

### arguments
The Spryk argument list. These arguments are used in the Spryk builder. See below for more detailed description.

### preSpryks
The Spryks that will be executed before the current Spryk.

### postSpryks
The Spryks that should be executed after the current Spryk.

### excludedSpryks

> Exclude the execution of the Spryks that are placed is `preSpryks` and `postSpryks`. Useful when you reuse the Spryk and you don't need some Spryks that are defined in `preSpryks` or `postSpryks`.

Reviewer: ```suggestion
Excludes the execution of the Spryks that are placed is `preSpryks` and `postSpryks`. Useful when you reuse a Spryk and you don't need some Spryks that are defined in `preSpryks` or `postSpryks`.
```

### preCommands
The commands that should be executed before the current Spryk.

### postCommands
The commands that should be executed after the current Spryk.

## Arguments

```yaml
arguments:
    organization: # argument name
        inherit: true
        default: Spryker
```

### inherit
Declares that the argument value can be inherited from the parent Spryk when not set explicitly.

### default
The default value for the argument if not passed from CLI.

### value

> The argument value that will be used. Useful when need to compose value from another arguments values or/and apply some twig filters or functions.

Reviewer: ```suggestion
The argument value to be used. Useful when you need to compose a value from another argument values or apply some twig filters or functions.
```

```yaml
arguments:
    target:
      value: "{{ organization }}\\Glue\\{{ module }}\\Dependency\\Client\\{{ module }}To{{ dependentModule }}ClientBridge"

    name:
      value: "CLIENT_{{ dependentModule | underscored | upper }}"
```

### isOptional

> If argument is optional the values can be empty otherwise the argument value should be provided. This option is `false` by default which means the value is required.

Reviewer: ```suggestion
If argument is optional, the values can be empty. Otherwise, the argument value should be provided. This option is `false` by default, which means the value is required.
```

### isMultiple
The argument can have multiply values and value can be provided as list of values.
```yaml
arguments:
    target:
        isMultiple: true
```

### allowOverride

> This option is only valid for the body argument of method Spryk and define you can or not to override existing method body.

Reviewer: ```suggestion
This option is only valid for the body argument of a Spryk method and defines whether you can override the existing method body.
```

```yaml
body:
  allowOverride: true
  value: "App/Registry/ZedControllerDisconnectMethod.php.twig"
```

### callback

> The pre-processing callback that will be applied on value before passing into the Spryk. Should implement `SprykerSdk\Spryk\Model\Spryk\Definition\Argument\Callback\CallbackInterface`.

Reviewer: ```suggestion
The pre-processing callback that will be applied on value before passing to the Spryk. It should implement `SprykerSdk\Spryk\Model\Spryk\Definition\Argument\Callback\CallbackInterface`.
```
